[Tooling] Migrate Prototype Builds from App Center to Firebase App Distribution

AliSoftware opened the pull request:
## Description

This migrates the Prototype Builds to be uploaded to Firebase App Distribution instead of App Center.

This is done in the context of App Center shutting down at the end of the month.

### Internal references:

 - pdnsEh-1zD-p2 
 - paaHJt-7S2-p2
 - 186-gh-Automattic/apps-infra-plans

### Release Timing

> [!IMPORTANT]
> We might want to wait before merging this until our internal documentation—about how to use our internal tools to enroll to Firebase App Distribution—has been updated.

## Testing

 - Check that the information shown in the PR comment looks accurate, including the build number (expected to be set to the Buildkite build number) and version (expected to use the PR number)
 - Add yourself to the appropriate Testing Group via our MC tool (`/mobile/firebase-app-distribution/testers/`), then accept the invitation email you received
 - Scan the QR code in the PR comment with your iPhone and confirm you can install the prototype build on your device

AliSoftware committed:
Update gems

AliSoftware committed:
Update Fastfile to use FAD

AliSoftware committed:
Remove remaining references to App Center

dangermattic commented:
<!--
  0 Errors
  0 Warnings
  1 Message: This PR is still a Draft: some...
  0 Markdowns
-->
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR is still a Draft: some checks will be skipped.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

AliSoftware committed:
Simplify pipeline logic

AliSoftware committed:
Fix early return logic

AliSoftware committed:
Provide ipa_path explicitly

Since it was build in a separate job so `firebase_app_distribution` can't infer it from `gym`'s `lane_context`

pocketcasts commented:
<!-- REUSE_ID: prototype-build-link -->

<details><summary><img align='top' src='https://raw.githubusercontent.com/buildkite/emojis/main/img-buildkite-64/firebase.png' width='20px' alt='App Icon' />📲 You can test the changes from this Pull Request in <b>Pocket Casts Prototype Build</b> by scanning the QR code below to install the corresponding build.</summary>
<table>
<tr>
  <td rowspan='6' width='260px'><img src='https://api.qrserver.com/v1/create-qr-code/?size=500x500&qzone=4&data=https%3A%2F%2Fappdistribution.firebase.google.com%2Ftesterapps%2F1%3A124902176124%3Aios%3A3999f43bfa1fcd1b1620f9%2Freleases%2F008phbtj6lt4o%3Futm_source%3Dfastlane' width='250' height='250' /></td>
  <td><b>App Name</b></td><td>Pocket Casts Prototype Build</td>
</tr>
<tr><td><b>Build Number</b></td><td><code>9939</code></td></tr>
<tr><td><b>Version</b></td><td><code>PR #2857</code></td></tr>
<tr><td><b>Bundle ID</b></td><td><code>au.com.shiftyjelly.podcasts.prototype</code></td></tr>
<tr><td><b>Commit</b></td><td>01e51ed8161ab8e6b1d84ebba42b7ad63728109e</td></tr>
<tr><td><b>Installation URL</b></td><td><a href='https://appdistribution.firebase.google.com/testerapps/1:124902176124:ios:3999f43bfa1fcd1b1620f9/releases/008phbtj6lt4o?utm_source=fastlane'>008phbtj6lt4o</a></td></tr>
</table>
<em>Automatticians: You can use our internal self-serve MC tool to give yourself access to those builds if needed.</em>
</details>

AliSoftware committed:
Fix version fields inversion

AliSoftware committed:
Generate new Prototype Build

Dummy empty CI commit to trigger a new Prototype Build with a new version and build number, so I can screenshot the install page from my phone for documentation purposes while having that page show more than one build (latest + previous ones)

twstokes reviewed (approved):
LGTM 🚀

SergioEstevao reviewed (approved):
The MC tool didn't work correctly in order to add me to the Firebase Group. But after I was added manually all worked correctly! 
App installed and working on my device!

AliSoftware committed:
Fix BUILDKITE_PULL_REQUEST being false not nil

AliSoftware committed:
Merge branch 'trunk' into tooling/migrate-from-appcenter-to-fad

# Conflicts:
#	Gemfile.lock
